app/react/Layout/specs/DocumentsList.spec.js

```diff
@@ -90,6 +91,16 @@ describe('DocumentsList', () => {
   describe('Graph view', () => {
     beforeEach(() => {
       props.view = 'graph';
+      props.connections = { totalRows: 2 };
+      props.connectionsGroups = Immutable.fromJS([
+        Immutable.fromJS({
+          templates: Immutable.fromJS([
+            {
+              count: Immutable.fromJS(2),
+            },
+          ]),
+        }),
+      ]);
```

Comment:
Immutable already works recursively.  `Immutable.fromJS([{ a: 'b' }])` will create an instance of Immutable.Map on that first element.  There is no need to create immutables from the children.  Even more so with numeric values... you do not need to make `2` an immutable!

Reply:
That's an ugly one. Getting to it!